WordPress/src/main/java/org/wordpress/android/ui/media/services/MediaUploadService.java

```diff
@@ -117,12 +130,10 @@ private void handleOnMediaUploadedSuccess(@NonNull OnMediaUploaded event) {
         } else if (event.completed) {
             // Upload completed
             AppLog.i(T.MEDIA, "Upload completed - localId=" + event.media.getId() + " title=" + event.media.getTitle());
-
-            // here we need to edit the corresponding post
-            updatePostWithMediaUrl(event.media);
             trackUploadMediaEvents(AnalyticsTracker.Stat.MEDIA_UPLOAD_SUCCESS, getMediaFromQueueById(event.media.getId()), null);
-
             completeUploadWithId(event.media.getId());
+            // add the MediaModel object within the OnMediaUploaded event to our completedMediaList
+            addMediaToPostCompletedMediaListMap(event.media);
```

Comment:
This gets called even for single-media uploads from the library, with the result that we're adding medial lists to the completed upload map with post ID `0`.